src/storage/blockchain/state.ts

```diff
@@ -102,29 +102,14 @@ export class EthStateStorage implements IStateStorage {
   }
 
   /** {@inheritdoc IStateStorage.publishState} */
-  async publishState(proof: ZKProof, signer: Signer): Promise<string> {
-    const stateTransitionPubSig = new StateTransitionPubSignals();
-    stateTransitionPubSig.pubSignalsUnmarshal(
-      byteEncoder.encode(JSON.stringify(proof.pub_signals))
-    );
-    const { userId, oldUserState, newUserState, isOldStateGenesis } = stateTransitionPubSig;
-
+  async publishState(
+    proof: ZKProof,
+    signer: Signer,
+    userStateTranstion: UserStateTransition
```

Comment:
we can't just break function API, let's create a new function publishStateGeneric(methodId, userStateTranstionInfo, calldata : hexstring)
which is corresponding to https://github.com/iden3/contracts/blob/2d38d39b65ce3d254f9c26f34c3f6010b19382a6/contracts/state/State.sol#L139
and use it

Reply:
Created function  `publishStateGeneric(signer, userStateTransitionInfo)` and  `methodId` and `methodParams` added inside the interface `UserStateTransitionInfo` like this
```
export interface UserStateTransitionInfo {
  userId: Id; // Identity id
  oldUserState: Hash; // Previous identity state
  newUserState: Hash; // New identity state
  isOldStateGenesis: boolean; // Is the previous state genesis?
  methodId: bigint; // State transition method id
  methodParams: string; // State transition method-specific params
}
```